### PR TITLE
fix: when click clear button, it will be focused

### PR DESCRIPTION
**Pull request by Pulset**
fix https://github.com/ant-design/ant-design/issues/36048

### CODE REVIEW

**Comment by Pulset** [comment]
@zombieJ Help review the code

File: src/BaseSelect.tsx [hunk 1]
```diff
@@ -691,6 +691,10 @@ const BaseSelect = React.forwardRef((props: BaseSelectProps, ref: React.Ref<Base
   const onClearMouseDown: React.MouseEventHandler<HTMLSpanElement> = () => {
     onClear?.();
 
+    selectorRef.current?.focus();
+    setMockFocused(true);
```
Comment (zombieJ) [inline src/BaseSelect.tsx:695]: focus will trigger `onFocus` event which will call `setMockFocused(true)`, no need manually call here and follow code also should be auto.

File: tests/Select.test.tsx [hunk 2]
```diff
@@ -1949,4 +1949,26 @@ describe('Select.Basic', () => {
...
+  it('should be focused when click clear button', () => {
+    jest.useFakeTimers();
+
+    const mouseDownPreventDefault = jest.fn();
+    const wrapper = mount(
+      <Select allowClear value={1}>
+        <Option value="1">1</Option>
+        <Option value="2">2</Option>
+      </Select>,
+    );
```
Comment (zombieJ) [inline tests/Select.test.tsx:1962]: should attchTo body since enzyme not fire `onFocus` if node is not in the document

**Comment by zombieJ** [comment]
`+ rc-select@14.1.16`

**Pull request merged**